test/src/folding/parsers/python_parser_test.dart

```diff
@@ -0,0 +1,207 @@
+            _FB(startLine: 3, endLine: 6, type: FoldableBlockType.brackets),
+            _FB(startLine: 7, endLine: 10, type: FoldableBlockType.brackets),
+          ],
+        ),
+        //
+        _Example(
+          'Python. Single-line comments',
+          code: '''
+class Mapping:
+    def __init__(self, iterable):
+        a = [
+            [
+              # comment
+              # another comment
+              # third comment
+                5,
+                6
+            ],
+            [
+                7,
+                8
+            ]
+        ]''',
+          expected: [
+            _FB(startLine: 0, endLine: 14, type: FoldableBlockType.indent),
+            _FB(startLine: 1, endLine: 14, type: FoldableBlockType.indent),
+            _FB(startLine: 2, endLine: 14, type: FoldableBlockType.brackets),
+            _FB(startLine: 3, endLine: 9, type: FoldableBlockType.brackets),
+            _FB(
+              startLine: 4,
+              endLine: 6,
+              type: FoldableBlockType.singleLineComment,
+            ),
+            _FB(startLine: 10, endLine: 13, type: FoldableBlockType.brackets),
+          ],
+        ),
+        //
+        _Example(
+          'Python. Pair characters in literals are ignored',
+          code: '''
+a = "{[(";
+b = ")]}";
+''',
+          expected: [],
+        ),
+        //
+        _Example(
+          'Python. Named comments',
+          code: '''
+class Mapping:
+    def __init__(self, iterable):
+        # [START section1]
+        # [END section1]
+        a = 5''',
+          expected: [
+            _FB(startLine: 0, endLine: 4, type: FoldableBlockType.indent),
+            _FB(startLine: 1, endLine: 4, type: FoldableBlockType.indent),
+          ],
+        ),
+      ];
+
+      for (final example in examples) {
+        highlight.registerLanguage('language', python);
+        final highlighted = highlight.parse(example.code, language: 'language');
+
+        final sequences = SingleLineComments.byMode[python] ?? [];
+
+        final commentParser = SingleLineCommentParser.parseHighlighted(
+          text: example.code,
+          highlighted: highlighted,
+          singleLineCommentSequences: sequences,
+        );
+
+        final serviceComments = ServiceCommentFilter.filter(
+          commentParser.comments,
+          namedSectionParser: const BracketsStartEndNamedSectionParser(),
+        );
+
+        final codeLines = CodeLineBuilder.textToCodeLines(
+          text: example.code,
+          highlighted: highlighted,
+          commentsByLines: commentParser.getCommentsByLines(),
+        );
+
+        final pythonParser = PythonFoldableBlockParser().parse(highlighted,
+            serviceComments.map((e) => e.source).toSet(), codeLines);
+
+        expect(
+          pythonParser,
+          example.expected,
+          reason: '${example.name}, valid blocks',
+        );
+      }
+    });
+  });
```

Comment:
When composing examples, please add diversity:
- Add comments after some lines that have `:` as their last significant character.
- Use trailing commas on some but not all multiline lists.
- Add global nodes other than `class`: variables, functions, etc.
- Use one-liners instead of some blocks like `if True: continue` to make sure they do not form blocks.